_dprhtml/features/navigation/init.js

```diff
@@ -1,206 +1,232 @@
+        onmouseup: 'DPRSend.importXML(false,null,null,null,DPRSend.eventSend(event))',
+      }
+  }
+
+  static isStorageSupportedByBrowser() {
+    return typeof Storage !== 'undefined'
+  }
+
+  async sendSelectedHistoryItem(ctx) {
+    if (ctx.selectedHistoryItem() && ctx.selectedHistoryItem() !== '-- History --') {
+      const selectedHistItem = ctx.selectedHistoryItem().toString().replace(/'/g, '').split('@')
+      const x = selectedHistItem[1].split(',')
+      if (x.length > 3) {
+        await DPRSend.openPlace(this.sectionId, x)
+      } else {
+        await DPRSend.openIndex(this.sectionId, x)
+      }
+    }
+  }
+
+  static computeHistoryInfo() {
+    return {
+      text: '\u21D2',
+      title: 'Open bookmarks and history window',
+      onmouseup: 'window.DPR_bookmarks_mod.bookmarkframe(1)',
+    }
+  }
+
+  updateHistory() {
+    if (NavigationTabViewModel.isStorageSupportedByBrowser()) {
+      if (!localStorage.getItem('navHistoryArray')) {
+        localStorage.setItem('navHistoryArray', JSON.stringify(['-- History --']))
+      }
+      this.navHistoryArray(JSON.parse(localStorage.getItem('navHistoryArray')))
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  sendSelectedBookmarksItem(ctx) {
+    // eslint-disable-next-line quotes
+    if (ctx.selectedBookmarksItem() && ctx.selectedBookmarksItem() !== "-- Bookmarks --") {
+      const selectedBookmItem = ctx.selectedBookmarksItem().toString().replace(/'/g, '').split('@')
+      const x = selectedBookmItem[1].split(',')
+      const sectionId = window.DPR_Chrome.getPrimarySectionId()
+      return x.length > 3 ? DPRSend.openPlace(sectionId, x) : DPRSend.openIndex(sectionId, x)
+    }
+
+    return Promise.resolve()
+  }
+
+  static computeBookmarksInfo() {
+    return {
+      text: '\u21D2',
+      title: 'Open bookmarks and history window',
+      onmouseup: 'window.DPR_bookmarks_mod.bookmarkframe(1)',
+    }
+  }
+
+  updateBookmarks() {
+    if (NavigationTabViewModel.isStorageSupportedByBrowser()) {
+      if (!localStorage.getItem('bookmarksArray')) {
+        localStorage.setItem('bookmarksArray', JSON.stringify(['-- Bookmarks --']))
+      }
+      this.bookmarksArray(JSON.parse(localStorage.getItem('bookmarksArray')))
+    }
+  }
+}
+
+export const ViewModel = new NavigationTabViewModel()
+DprGlobals.singleton.NavigationTabViewModel = ViewModel
+
+export const initializeFeature = async (sectionId) => {
+  await window.DPR_config_mod.getconfig()
+  await window.DPR_Chrome.addMainPanelSections(
+    ViewModel.places(),
+    sectionId,
+    ViewModel.query(),
+    ViewModel.para(),
+  )
+}
+
+const parseNavigationURLParams = () => {
+  const urlParams = decodeURIComponent(window.location.search).substring(1, window.location.search.length).split('&')
+  let query = ''
+  let para = ''
+  urlParams.forEach((parameter) => {
+    const [psec0, psec1, ..._] = parameter.split('=')
```

Comment:
If you wanted to eliminate a little bit of ​noise, I believe the following is semantically equivalent:
```suggestion
    ​const [psec0, psec1] = parameter.split('=')
``

Reply:
fixed